developer-workflow/c-api.rst

```diff
@@ -122,6 +122,32 @@ Guidelines for expanding/changing the public API
   - ``return 0``: lookup succeeded; no item was found
   - ``return 1``: lookup succeeded; item was found
 
+- APIs with output parameters should ensure that each output parameter is
+  initialised for all code paths.
+  Example:
+
+  .. code-block:: c
+
+     int
+     PyFoo_Bar(PyObject **out)
+     {
+         PyObject *value;
+         int rc = foo_bar(&value);
```

Comment:
Maybe you could mimick a legacy API call which returns NULL if not found *and* on error, and call PyErr_Occurred(). Sot he difference with the rwo APIs is even more obvious?